phac-nml/irida-next
# Sample history view

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in a History page for samples which displays the record level changes made to the sample (name, description, project id, metadata). The versions are displayed in a timeline view and clicking on the version number displays a modal with which attributes were changed and by whom.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/f59e388b-6a4d-4b0b-8ad3-797b1df55565)

![image](https://github.com/phac-nml/irida-next/assets/25466211/7efffda0-c951-4cf1-8620-b14c189b1a6f)

![image](https://github.com/phac-nml/irida-next/assets/25466211/3e69c144-1ec6-4e15-b172-822cfb449ef9)

![image](https://github.com/phac-nml/irida-next/assets/25466211/932146ef-4ddc-4b9e-a581-af1d647e8922)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Reset db (drop, create, migrate, seed)
2. Login as a user
3. Create a project
4. Create a sample in the project
5. View the sample and click on the `History` tab for the sample 
6. You should see 1 version, and clicking the `Version 1` should launch a modal that displays the attributes which the sample was created with
7. Edit the sample by changing the name
8. On the `History` tab for the sample you should now see 2 versions. Clicking the `Version 2` should launch a modal which displays the change to the name attribute. On the left hand side is the previous version of that attribute and on the right is the current version
9. Try the same for description, adding new metadata, removing metadata, and by transferring the sample to another project (you will need to view the sample in the project it was transferred to)
10. Also try to view history for other samples that were seeded in projects that the user has direct membership or inherited membership on (shared group members should not be able to see the history for a sample)


## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
